actionpack/test/controller/caching_test.rb

```diff
@@ -406,4 +419,25 @@ def test_caching_works_with_beginning_comment
     ActionView::PartialRenderer.expects(:collection_with_template).never
     get :index_with_comment
   end
+
+  def test_caching_works_with_collection_rendering_multiple_templates
+    get :index_with_multiple_templates
+
+    ActionView::PartialRenderer.expects(:collection_without_template).never
```

Comment:
I agree, it's terrible.

Internally `render_collection` calls `collection_without_template` to actually render the templates. I couldn't think of a better way to test that no new templates were rendered.

(I'm also a bit wary of mocks and stubs as it feels like there's a greater chance of false positives there to me.)